packages/notices/src/store/actions.js

```diff
@@ -313,3 +313,48 @@ export function removeNotice( id, context = DEFAULT_CONTEXT ) {
 		context,
 	};
 }
+
+/**
+ * Removes all notices from a given context. Defaults to the default context.
+ *
+ * @param {string} context The context to remove all notices from.
+ *
+ * @example
+ * ```js
+ * import { __ } from '@wordpress/i18n';
+ * import { useDispatch, useSelect } from '@wordpress/data';
+ * import { store as noticesStore } from '@wordpress/notices';
+ * import { Button } from '@wordpress/components';
+ *
+ * export const ExampleComponent = () => {
+ * 	const notices = useSelect( ( select ) =>
+ * 		select( noticesStore ).getNotices()
+ * 	);
+ * 	const { removeNotices } = useDispatch( noticesStore );
+ * 	return (
+ * 		<>
+ * 			<ul>
+ * 				{ notices.map( ( notice ) => (
+ * 					<li key={ notice.id }>{ notice.content }</li>
+ * 				) ) }
+ * 			</ul>
+ * 			<Button
+ * 				onClick={ () =>
+ * 					removeAllNotices()
+ * 				}
+ * 			>
+ * 				{ __( 'Clear all notices', 'woo-gutenberg-products-block' ) }
+ * 			</Button>
+ * 		</>
+ * 	);
+ * };
+ * ```
+ *
+ * @return {Object} 	   Action object.
+ */
+export function removeAllNotices( context = DEFAULT_CONTEXT ) {
```

Comment:
One thing about this API is calling it would remove both 'snackbar' notices and also notices that have a `type` of 'default'.

From a UI point of view, it might look unusual to see two things that are visually different suddenly disappear from a single dismissal, so I wonder if there should be a way to filter by `type` as well. The same feedback also applies to #39940.

What do you think?

Something that would be helpful is an example of the use case this PR is fulfilling. https://github.com/WordPress/gutenberg/pull/39940 shows the developer requirement, but doesn't describe the feature this API supports from a user perspective (if there is one).

Reply:
@talldan thanks for the review - I agree that allowing filtering by type is a great idea. I will implement this in both PRs.

The difference between this and #39940 is this one does not require any IDs to be passed to the action which will confer a slight performance improvement based on selecting fewer values from the store and dispatching fewer actions.

A use-case I had in mind for this would be if an app is showing multiple notices, and there is the option to clear all notices, then currently they'd need to loop through the notices and get each ID, then remove each individually. Admittedly the user experience is no different if all notices are removed in one action, or _n_ actions, the end-result is the same.

Reply:
I'm not convinced that filtering by type in #39940 is a good idea - the developer would have to already know the IDs to remove them, so asking them to pass the type as well seems like it would add more friction for not much gain. I know a snackbar notice and default notice can both have the same ID, but `removeNotice` does not ask for a type, so when using this it would remove notices of both type anyway.